django_geosource/models.py

```diff
@@ -373,3 +374,148 @@ def refresh_data(self):
+                )
+            except pyexcel.exceptions.FileTypeNotSupported as err:
+                msg = "Source's CSV file is not valid"
+                logger.info(msg)
+                err.message = msg  # new message for the user
+                raise
+
+    def _get_records(self, limit=None):
+        sheet = self.get_file_as_sheet()
+        if self.settings.get("header"):
+            sheet.name_columns_by_row(0)
+
+        limit = limit if limit else len(sheet)
+
+        records = []
+        srid = self.settings["coordinate_reference_system"]
+        for row in sheet:
+            if self.settings["coordinates_field"] == "two_columns":
+                lat_field = self.settings["latitude_field"]
+                lng_field = self.settings["longitude_field"]
+
+                # if no header, we except index for the columns has been provided
+                if self.settings.get("header"):
+                    x = row[sheet.colnames.index(lng_field)]
+                    y = row[sheet.colnames.index(lat_field)]
+                else:
+                    x = row[sheet.colnames[int(lng_field)]]
+                    y = row[sheet.colnames[int(lat_field)]]
+
+                records.append(
+                    {
+                        self.SOURCE_GEOM_ATTRIBUTE: GEOSGeometry(
+                            f"Point({x} {y})", srid=srid
+                        ),
+                        **{
+                            name: value
+                            for i, (name, value) in enumerate(zip(sheet.colnames, row))
+                            if i not in (row.index(x), row.index(y))
+                        },
+                    }
+                )
+            else:
+                sep = self.settings["coordinates_separtor"]
+                is_xy = self.settings["coordinates_field_count"] == "xy"
+                lnglat_field = self.settings["latlong_field"]
+
+                # if no header, we except index for the column has been provided
+                if self.settings.get("header"):
+                    coords = row[sheet.colnames.index(lnglat_field)]
+                else:
+                    coords = row[sheet.colnames[int(lnglat_field)]]
+
+                # some fools use a reversed cartesian coordinates system (╯°□°)╯︵ ┻━┻
+                x, y = coords.split(sep) if is_xy else coords.split(sep).reverse()
+                records.append(
+                    {
+                        self.SOURCE_GEOM_ATTRIBUTE: GEOSGeometry(
+                            f"Point({x} {y})", srid=srid
+                        ),
+                        **{
+                            name: value
+                            for i, (name, value) in enumerate(zip(sheet.colnames, row))
+                            if i != row.index(coords)
+                        },
+                    }
+                )
+
+        return records
+
+    def _get_separator(self, name):
+        SEPARATORS = {
```

Comment:
separator should be declared at upper level.

Reply:
Do you mean at upper class level or file level?

Reply:
I put `SEPARATORS` at upper level in class.